edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/Program.cs

```diff
@@ -235,6 +240,32 @@ public static async Task<int> MainAsync(IConfiguration configuration)
                 return 1;
             }
 
+            // Initialize metrics
+            var metricsConfig = new MetricsConfig(true, MetricsListenerConfig.Create(configuration));
+            var metricsProvider = metricsConfig.Enabled
+                             ? new MetricsProvider("edgeagent", iothubHostname, deviceId)
+                             : new NullMetricsProvider() as IMetricsProvider;
+            var metricsListener = metricsConfig.Enabled
+                            ? new MetricsListener(metricsConfig.ListenerConfig, metricsProvider)
+                            : new NullMetricsListener() as IMetricsListener;
+            Metrics.Init(metricsProvider, metricsListener, logger);
+
+            // Initialize exceptions counter
+            Dictionary<Type, string> recognizedExceptions = new Dictionary<Type, string>
+            {
+                // TODO: Decide what exceptions to recognize and ignore
+                { typeof(Newtonsoft.Json.JsonSerializationException), "json_serialization" },
+                { typeof(ArgumentException), "argument" },
+                // { typeof(Docker.DotNet.DockerApiException), "docker" },
+                { typeof(Rest.HttpOperationException), "http" },
+            };
+            HashSet<Type> ignoredExceptions = new HashSet<Type>
+            {
+                typeof(TaskCanceledException),
+                typeof(OperationCanceledException),
+            };
+            new ExceptionCounter(recognizedExceptions, ignoredExceptions);
```

Comment:
It's lifetime is until the program ends, so it shouldn't be necessary.

Reply:
If there's no variable, then couldn't the garbage collector destroy the memory from this `new` anytime it wants?

Reply:
I probably should capture and destroy it just to be explicit, but since it attaches one of its members to an event emitter, it will never get garbage collected unless it unsubscribes (which it's destructor does)